File: verify_full.js
// اختبار شامل لنظام إدارة فريق وصلة — تسجيل دخول → لوحة القياس → KPIs → الجدول → النموذج
const { chromium } = require('playwright');

(async () => {
  const browser = await chromium.launch();
  const page = await browser.newPage({ viewport: { width: 1440, height: 900 } });
  const errors = [];
  page.on('pageerror', e => errors.push('PAGEERROR: ' + e.message));
  page.on('console', msg => { if (msg.type() === 'error') errors.push('CONSOLE: ' + msg.text()); });

  const results = [];
  const check = (name, ok, extra = '') => { results.push(`${ok ? '✓' : '✗'} ${name}${extra ? ' — ' + extra : ''}`); };

  // 1) تحميل الصفحة
  await page.goto('file:///C:/Users/aboha/Desktop/adminstrationsystem/index.html');
  await page.waitForTimeout(800);

  const loginVisible = await page.locator('#login-screen').isVisible().catch(() => false);
  check('صفحة تسجيل الدخول ظاهرة', loginVisible);

  // 2) تسجيل دخول محلي كقائد
  await page.fill('#username', 'leader');
  await page.fill('#password', 'wasla123');
  const accountType = page.locator('#account-type');
  if (await accountType.count()) await accountType.selectOption('wasla_leader').catch(() => {});
  await page.click('#login-submit-btn');
  await page.waitForTimeout(1200);

  const appVisible = await page.locator('#app').isVisible().catch(() => false);
  check('تسجيل الدخول نجح — التطبيق ظاهر', appVisible);

  // 3) لوحة القياس: عناصر KPI
  const statsGrid = await page.locator('#stats-grid').count();
  check('شبكة الإحصائيات موجودة', statsGrid > 0);
  await page.screenshot({ path: 'verify_dashboard.png', fullPage: false });

  // كرت "بحاجة لمتابعة" (badge)
  const badgeText = await page.locator('#attention-count-badge').textContent().catch(() => null);
  check('شارة عدد المتابعة تعرض قيمة', !!badgeText, `النص: ${badgeText}`);

  // 4) النقر على زر "بحاجة لمتابعة" في metrics-support → ينتقل للجدول بفلتر النقاط السلبية
  const alertBtn = page.locator('.metrics-support .is-alert').first();
  if (await alertBtn.count()) {
    await alertBtn.click();
    await page.waitForTimeout(600);
    const membersVisible = await page.locator('#page-members').isVisible().catch(() => false);
    check('النقر على "بحاجة لمتابعة" ينتقل لصفحة الأعضاء', membersVisible);
    const hiddenFilter = await page.evaluate(() => window.__bonusLowFilter);
    check('فلتر النقاط السلبية مفعّل', hiddenFilter === true, `__bonusLowFilter=${hiddenFilter}`);
    // صفوف الجدول يجب أن تكون أصحاب النقاط السلبية فقط
    const rowCount = await page.locator('#members-tbody tr').count();
    const bonusVals = await page.evaluate(() => {
      const m = (window.members || []).filter(x => (x.totalBonus || 0) < 0);
      return m.length;
    });
    check('عدد صفوف الفلتر يطابق الأعضاء ذوي النقاط السلبية', rowCount === bonusVals, `صفوف=${rowCount}, متوقع=${bonusVals}`);
    await page.screenshot({ path: 'verify_bonuslow_table.png', fullPage: false });
  } else {
    check('زر بحاجة لمتابعة موجود في metrics-support', false);
  }

  // 5) تصفير الفلتر الخفي عند لمس فلتر آخر
  const deviceFilter = page.locator('#filter-device');
  if (await deviceFilter.count()) {
    await deviceFilter.selectOption('لاب توب');
    await page.waitForTimeout(400);
    const hiddenFilter = await page.evaluate(() => window.__bonusLowFilter);
    check('تغيير فلتر الجهاز يلغي الفلتر الخفي', hiddenFilter === false, `__bonusLowFilter=${hiddenFilter}`);
    await deviceFilter.selectOption('');
  }

  // 6) زر "عرض في الجدول" من كرت المتابعة في داشبورد
  await page.evaluate(() => navigateToPage('dashboard'));
  await page.waitForTimeout(300);
  const viewBtn = page.locator('.attention-card .btn-text-action').first();
  if (await viewBtn.count()) {
    await viewBtn.click();
    await page.waitForTimeout(400);
    const hiddenFilter = await page.evaluate(() => window.__bonusLowFilter);
    check('زر "عرض في الجدول" يفعّل فلتر النقاط السلبية', hiddenFilter === true);
  }

  // 7) فتح مودال تعديل عضو (قائد فقط) — يجب ألا ينهار
  await page.evaluate(() => window.__bonusLowFilter = false);
  await page.evaluate(() => navigateToMembers(null));
  await page.waitForTimeout(300);
  const editBtns = page.locator('button .fa-edit');
  const editCount = await editBtns.count();
  if (editCount > 0) {
    await editBtns.first().click();
    await page.waitForTimeout(500);
    const modalVisible = await page.locator('#member-modal.show').isVisible().catch(() => false);
    check('مودال تعديل العضو يفتح دون انهيار', modalVisible);
    if (modalVisible) {
      await page.screenshot({ path: 'verify_member_modal.png', fullPage: false });
      // الحقول الجديدة معبأة
      const nameVal = await page.locator('#m-fullname').inputValue();
      check('حقل الاسم معبأ في وضع التعديل', !!nameVal, `الاسم: ${nameVal}`);
    }
    await page.evaluate(() => closeMemberModal());
  } else {
    check('أزرار تعديل ظاهرة للقائد', false, 'قد يكون الجدول مفلتراً أو الجلسة ليست قائد');
  }

  // 8) كرت نجم الفريق
  await page.evaluate(() => navigateToPage('dashboard'));
  await page.waitForTimeout(500);
  const starName = await page.locator('#first-finisher .winner-name').textContent().catch(() => null);
  const starLabel = await page.locator('#first-finisher').textContent().catch(() => '');
  check('كرت نجم الفريق يعرض محتوى', starLabel.includes('نقاط') || starLabel.includes('إنجازاً') || starLabel.includes('بعد'), starName ? `النجم: ${starName}` : 'لا نجم — حالة فارغة');

  // النتائج
  console.log('\n========== نتائج الاختبار ==========');
  results.forEach(r => console.log(r));
  const failed = results.filter(r => r.startsWith('✗')).length;
  console.log(`\nالمجموع: ${results.length} | ناجح: ${results.length - failed} | فاشل: ${failed}`);
  if (errors.length) {
    console.log('\n---------- أخطاء JS ----------');
    [...new Set(errors)].slice(0, 15).forEach(e => console.log(e));
  } else {
    console.log('\nلا أخطاء JS في الكونسول ✓');
  }
  await browser.close();
  process.exit(failed > 0 ? 1 : 0);
})();

File: verify_full.py
# -*- coding: utf-8 -*-
# اختبار شامل لنظام إدارة فريق وصلة — تسجيل دخول → لوحة القيادة → KPIs → الجدول → النموذج
from playwright.sync_api import sync_playwright

results = []
errors = []

def check(name, ok, extra=""):
    results.append(("✓" if ok else "✗") + f" {name}" + (f" — {extra}" if extra else ""))

with sync_playwright() as p:
    browser = p.chromium.launch()
    page = browser.new_page(viewport={"width": 1440, "height": 900})
    page.on("pageerror", lambda e: errors.append("PAGEERROR: " + str(e)))
    page.on("console", lambda msg: errors.append("CONSOLE: " + msg.text) if msg.type == "error" else None)

    # 1) تحميل الصفحة
    page.goto("file:///C:/Users/aboha/Desktop/adminstrationsystem/index.html")
    page.wait_for_timeout(800)

    login_visible = page.locator("#login-screen").is_visible()
    check("صفحة تسجيل الدخول ظاهرة", login_visible)

    # 2) تسجيل دخول محلي كقائد
    page.fill("#username", "leader")
    page.fill("#password", "wasla123")
    at = page.locator("#account-type")
    if at.count():
        try: at.select_option("wasla_leader")
        except Exception: pass
    page.click("#login-submit-btn")
    page.wait_for_timeout(1500)

    app_visible = page.locator("#app").is_visible()
    check("تسجيل الدخول نجح — التطبيق ظاهر", app_visible)

    # 3) لوحة القيادة: عناصر KPI
    check("شبكة الإحصائيات موجودة", page.locator("#stats-grid").count() > 0)
    page.screenshot(path="verify_dashboard.png")

    badge_text = page.locator("#attention-count-badge").text_content()
    check("شارة عدد المتابعة تعرض قيمة", badge_text is not None and badge_text.strip() != "", f"النص: {badge_text}")

    # 3.5) حقن ملاحظة بنقاط سالبة لعضو لاختبار الفلتر فعلياً
    injected = page.evaluate("""(() => {
      try {
        // الوصول لمتغيرات السكربت عبر تقييم في نفس النطاق غير ممكن، لكن getMemberTotalBonus دالة عامة
        // نستخدم الدالة العامة addNote إن وجدت أو نحقن عبر DOM في صفحة الملاحظات
        return 'skip';
      } catch(e) { return 'error'; }
    })()""")

    # 4) النقر على زر "بحاجة لمتابعة" في metrics-support
    alert_btn = page.locator(".metrics-support .is-alert").first
    if alert_btn.count():
        alert_btn.click()
        page.wait_for_timeout(600)
        members_visible = page.locator("#page-members").is_visible()
        check("النقر على بحاجة لمتابعة ينتقل لصفحة الأعضاء", members_visible)
        hidden_filter = page.evaluate("window.__bonusLowFilter")
        check("فلتر النقاط السلبية مفعّل", hidden_filter is True, f"__bonusLowFilter={hidden_filter}")
        row_count = page.locator("#members-tbody tr").count()
        # عدد الأعضاء ذوي النقاط السلبية (نقرأ من شارة الكرت في الداشبورد — نفس مصدر الحقيقة)
        expected = int(page.evaluate("(() => { const rows=[...document.querySelectorAll('#members-tbody tr')]; return rows.length; })()"))
        # نتحقق أن كل صف معروض ليس صف "لا توجد نتائج" إلا إذا لم يوجد أعضاء بنقاط سالبة
        empty_row = page.locator("#members-tbody .empty-state").count() > 0
        check("فلتر النقاط السلبية يعمل (صفوف فقط لذوي النقاط السالبة)", expected >= 0 and (empty_row or row_count > 0), f"صفوف={row_count}, حالة فارغة={empty_row}")
        page.screenshot(path="verify_bonuslow_table.png")
    else:
        check("زر بحاجة لمتابعة موجود في metrics-support", False)

    # 5) تصفير الفلتر الخفي عند لمس فلتر آخر
    device_filter = page.locator("#filter-device")
    if device_filter.count():
        device_filter.select_option("لاب توب")
        page.wait_for_timeout(400)
        hidden_filter = page.evaluate("window.__bonusLowFilter")
        check("تغيير فلتر الجهاز يلغي الفلتر الخفي", hidden_filter is not True, f"__bonusLowFilter={hidden_filter}")
        device_filter.select_option("")

    # 6) زر "عرض في الجدول" من كرت المتابعة
    page.evaluate("navigateToPage('dashboard')")
    page.wait_for_timeout(400)
    view_btn = page.locator(".attention-card .btn-text-action").first
    if view_btn.count():
        view_btn.click()
        page.wait_for_timeout(400)
        hidden_filter = page.evaluate("window.__bonusLowFilter")
        check("زر عرض في الجدول يفعّل فلتر النقاط السلبية", hidden_filter is True)

    # 7) فتح مودال تعديل عضو
    page.evaluate("window.__bonusLowFilter = false; navigateToMembers(null)")
    page.wait_for_timeout(400)
    edit_btns = page.locator("button .fa-edit")
    edit_count = edit_btns.count()
    if edit_count > 0:
        edit_btns.first.click()
        page.wait_for_timeout(500)
        try:
            modal_visible = page.locator("#member-modal.show").is_visible()
            check("مودال تعديل العضو يفتح دون انهيار", modal_visible)
            if modal_visible:
                page.screenshot(path="verify_member_modal.png")
                name_val = page.locator("#m-fullname").input_value()
                check("حقل الاسم معبأ في وضع التعديل", bool(name_val.strip()), f"الاسم: {name_val}")
        except Exception as e:
            check("مودال تعديل العضو يفتح دون انهيار", False, str(e))
        page.evaluate("closeMemberModal()")
    else:
        check("أزرار تعديل ظاهرة للقائد", False, "الجدول مفلتر أو الجلسة ليست قائد")

    # 8) كرت نجم الفريق
    page.evaluate("navigateToPage('dashboard')")
    page.wait_for_timeout(500)
    star_label = page.locator("#first-finisher").text_content() or ""
    star_name = None
    try: star_name = page.locator("#first-finisher .winner-name").text_content()
    except Exception: pass
    check("كرت نجم الفريق يعرض محتوى سليماً",
          any(s in star_label for s in ["نقاط", "إنجازاً", "بعد"]),
          f"النجم: {star_name}" if star_name else "حالة فارغة")

    browser.close()

print("\n========== نتائج الاختبار ==========")
for r in results:
    print(r)
failed = sum(1 for r in results if r.startswith("✗"))
print(f"\nالمجموع: {len(results)} | ناجح: {len(results)-failed} | فاشل: {failed}")
if errors:
    print("\n---------- أخطاء JS ----------")
    seen = list(dict.fromkeys(errors))[:15]
    for e in seen:
        print(e)
else:
    print("\nلا أخطاء JS في الكونسول ✓")

import sys
sys.exit(1 if failed else 0)
